packages/react/src/date-picker/DatePicker.tsx

```diff
@@ -85,20 +88,35 @@ export type DatePickerProps = DatePickerInputProps &
 	DatePickerBaseProps;
 
 export const DatePicker = ({
-	value,
-	onBlur: onBlurProp,
-	onChange,
-	onInputChange: onInputChangeProp,
-	minDate,
-	maxDate,
+	allowedDateFormats: allowedDateFormatsProp = acceptedDateFormats,
+	dateFormat = 'dd/MM/yyyy',
 	initialMonth,
-	yearRange,
 	inputRef,
 	invalid = false,
+	maxDate,
 	maxWidth = 'md',
-	dateFormat = 'dd/MM/yyyy',
+	minDate,
+	onBlur: onBlurProp,
+	onChange,
+	onInputChange: onInputChangeProp,
+	value,
+	yearRange,
 	...props
 }: DatePickerProps) => {
+	const allowedDateFormats = useMemo(
+		() =>
+			Array.from(
+				new Set([
+					// We need to ensure that the dateFormat to render is included in allowedDateFormats. It should also be the preferred format.
+					dateFormat,
+					...allowedDateFormatsProp.filter((dateFormat) =>
+						acceptedDateFormats.includes(dateFormat)
+					),
+				])
+			),
```

Comment:
Assuming the `filter` is to prevent disallowed date formats, you shouldn't need the `Set`:

```suggestion
		() => [
			// We need to ensure that the dateFormat to render is included in allowedDateFormats. It should also be the preferred format.
			dateFormat,
			...allowedDateFormatsProp.filter(
				(allowedDateFormat) =>
					dateFormat !== allowedDateFormat &&
					acceptedDateFormats.includes(allowedDateFormat)
			),
		],
```

And alternatively, if it's not to prevent disallowed date formats and just to de-dupe we could do this:

```suggestion
		() => Array.from(new Set([dateFormat, ...allowedDateFormatsProp])),
```

Reply:
Yeah, it's both. To prevent incorrect ones and to de-dupe if someone adds the `dateFormat` to the `allowedDateFormats`.